File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for ele in tokens:
            if ele not in '/*+-':
                stack.append(ele)
            else:
                right = int(stack.pop())
                left = int(stack.pop())

                if ele == "/":
                    stack.append(int(left/right))
                elif ele == "*":
                    stack.append(right * left)
                elif ele == "+":
                    stack.append(right + left)
                elif ele == "-":
                    stack.append(left - right)
            
        return int(stack.pop())

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function (tokens) {
    let stack = [];

    for (let ele of tokens) {
        if (!["+", "-", "*", "/"].includes(ele)) {
            stack.push(parseInt(ele));
        } else {
            let right = stack.pop();
            let left = stack.pop();

            if (ele === "/") {
                stack.push(Math.trunc(left / right)); // Ensures truncation towards zero
            } else if (ele === "*") {
                stack.push(left * right);
            } else if (ele === "-") {
                stack.push(left - right);
            } else if (ele === "+") {
                stack.push(left + right);
            }
        }
    }

    return stack.pop();
};